From 4c12b1ad4abc38e997d79a637580278833677363 Mon Sep 17 00:00:00 2001
From: Dongwon Kim <dongwon.kim@intel.com>
Date: Mon, 30 Sep 2024 18:41:55 -0700
Subject: [PATCH] ui/gtk: Hardcode default size of new tab window to 960 by 540

This is a WA to keep the display resolution from windows VM
after waking up from S4.

Signed-off-by: Dongwon Kim <dongwon.kim@intel.com>
---
 ui/console.c | 4 ++--
 ui/gtk.c     | 6 +++---
 2 files changed, 5 insertions(+), 5 deletions(-)

diff --git a/ui/console.c b/ui/console.c
index 5c8c55f0ca..e5de7055e4 100644
--- a/ui/console.c
+++ b/ui/console.c
@@ -1291,8 +1291,8 @@ QemuConsole *graphic_console_init(DeviceState *dev, uint32_t head,
 {
     static const char noinit[] =
         "Guest has not initialized the display (yet).";
-    int width = 640;
-    int height = 480;
+    int width = 960;
+    int height = 540;
     QemuConsole *s;
     DisplaySurface *surface;
 
diff --git a/ui/gtk.c b/ui/gtk.c
index c76e0c8536..4dfe99ca38 100644
--- a/ui/gtk.c
+++ b/ui/gtk.c
@@ -1695,12 +1695,12 @@ static void gd_vc_full_screen(void *opaque)
 static void gd_tab_window_create(VirtualConsole *vc)
 {
     GtkDisplayState *s = vc->s;
+    int default_width=960;
+    int default_height=540;
 
     gtk_widget_set_sensitive(vc->menu_item, false);
     vc->window = gtk_window_new(GTK_WINDOW_TOPLEVEL);
-    gtk_window_set_default_size(GTK_WINDOW(vc->window),
-                                surface_width(vc->gfx.ds),
-                                surface_height(vc->gfx.ds));
+    gtk_window_set_default_size(GTK_WINDOW(vc->window), default_width, default_height);
 #if defined(CONFIG_OPENGL)
     if (vc->gfx.esurface) {
         eglDestroySurface(qemu_egl_display, vc->gfx.esurface);
-- 
2.43.0

